assignment2/task1.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: UTF-8 -*-
+
+
+class AncestorsFinder(object):
+    def __init__(self, graph):
+        self.graph = graph
+        self.root = 0
+        self.ancestors = None
+
+    def find(self, value):
+        self.ancestors = []
+        self._dfs(self.root, -1, value)
+        return self.ancestors
+
+    def _dfs(self, ver, prev, value):
+        if self.graph[ver]['value'] == value:
+            return True
+        for to in self.graph[ver]['edges']:
+            if to == prev or not self._dfs(to, ver, value):
```

Comment:
If some edge go to previous vertex (from which we go to current), we skip this vertex (don't need to check one vertex twice). Otherwise for all `to` vertexes start new iteration of dfs method with parameters `self._dfs(to, ver, value)`. If descendants of `to` vertex do not contain `value`, go to next `to` and check it in the same way